### DISCUSSION TITLE
ERROR ibapi::client] error decoding message: error occurred: Invalid message length

### DISCUSSION THREAD

**Discussion by KeithKee** [wboayue/rust-ibapi#187] [General]
Many thanks again for sharing.

I get this error running the below example.

Running `target\debug\examples\readme_realtime_data_1.exe`
[2024-12-06T16:28:41Z ERROR ibapi::client] error decoding message: error occurred: Invalid message length

I have switched to MarketDataType::Delayed.

Please advise.

**Comment by wboayue** [comment]
I'll investigate. I use this API is my system, but not with delayed data.

**Comment by wboayue** [comment]
Try with `v1.0.4` of the library.

If you're still having an issue post the part of the log before the error. That should show the realtime bar message that it's trying to process.

**Comment by KeithKee** [comment]
git pull done - v1.0.4

 Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.50s
     Running `target\debug\examples\readme_realtime_data_1.exe`
market data switched: DelayedFrozen
[2024-12-08T15:48:07Z ERROR ibapi::client] error decoding message: parse error: 5 - Requested market data requires additional subscription for API. See link in 'Market Data Connections' dialog for more details. - invalid float literal

I am able to get delayed data from my  python script.

**Reply by wboayue** [reply]
With my account setup, I won't see this message. The fastest way to resolve this would be to look at the API messages in the TWS or Gateway logs. Compare and note the differences if message sent by the program when running the Python script vs Rust executable. With that data I can update the Rust side if needed.

**Reply by KeithKee** [reply]
Thanks for the effort. I thought I try another example.

PS C:\Users\keith\rust\rust-ibapi> cargo run --example tick_by_tick -- --bid_ask spy
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.06s
     Running `target\debug\examples\tick_by_tick.exe --bid_ask spy`
connection_string: 127.0.0.1:7497
market data switched: Delayed
[2024-12-10T15:18:54Z ERROR ibapi::client] error decoding message: error occurred: Unexpected tick_type: 9000
PS C:\Users\keith\rust\rust-ibapi> cargo run --example tick_by_tick -- --bid_ask spy
   Compiling ibapi v1.0.4 (C:\Users\keith\rust\rust-ibapi)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.74s
     Running `target\debug\examples\tick_by_tick.exe --bid_ask spy`
connection_string: 127.0.0.1:7497
market data switched: Live
[2024-12-10T15:19:28Z ERROR ibapi::client] error decoding message: error occurred: Unexpected tick_type: 9000

Is there a way, like interpreted Python, to have errors showing the code line where it stops or panics?

**Reply by wboayue** [reply]
Thanks for reporting this. I added an issue to update the tick by tick examples. The example code is using old contracts and not the symbol passed in as a command line argument. If you update the contract used in the code the example will work. I added an issue to cleanup the tick by tick examples.

**Reply by cometta** [reply]
> git pull done - v1.0.4
> 
> Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.50s Running `target\debug\examples\readme_realtime_data_1.exe` market data switched: DelayedFrozen [2024-12-08T15:48:07Z ERROR ibapi::client] error decoding message: parse error: 5 - Requested market data requires additional subscription for API. See link in 'Market Data Connections' dialog for more details. - invalid float literal
> 
> I am able to get delayed data from my python script.

may i know whether able to get 15mins delay data without subscription ibkr data? i get the same error `Requested market data requires additional subscription for API. See link in 'Market Data Connections' dialog for more details.--`
